cms/plugin_base.py

```diff
@@ -369,6 +374,18 @@ def save_model(self, request, obj, form, change):
 
         return super(CMSPluginBase, self).save_model(request, obj, form, change)
 
+    def save_form(self, request, form, change):
+        obj = super(CMSPluginBase, self).save_form(request, form, change)
+
+        if getattr(form, '_cms_initial_attributes', None):
+            # Form has the initial attribute hooks
+            obj.language = form._cms_plugin_language
+            obj.placeholder = form._cms_plugin_placeholder
+            obj.parent = form._cms_plugin_parent
+            obj.plugin_type = form._cms_plugin_plugin_type
+            obj.position = form._cms_plugin_position
+        return obj
+
```

Comment:
Perhaps this is a matter of style here, but `_cms_initial_attributes` seems like a superfluous flag. Perhaps instead of containing `True` or not existing at all, perhaps it should be a dict which contains all the other attributes, then you should be able to make these assignments more DRY'er, and the existence of the dict is now more meaningful. Something like:

```` python
def get_form(self, request, obj=None, **kwargs):
    form_class = super(CMSPluginBase, self).get_form(request, obj, **kwargs)

    if obj:
        return form_class

    plugin_data = self.validate_add_request(request)

    form_class._cms_initial_attributes = {
        'language': plugin_data['plugin_language'],
        'placeholder': plugin_data['placeholder_id'],
        'parent': plugin_data.get('plugin_parent', None),
        'plugin_type': plugin_data['plugin_type'],
        'position': plugin_data['position'],
    }
    return form_class


def save_form(self, request, form, change):
    obj = super(CMSPluginBase, self).save_form(request, form, change)

    if hasattr(form, '_cms_initial_attributes'):
        for key, value in form._cms_initial_attributes.items():
            setattr(obj, key, value)
    return obj
````